src/foam/dao/CopyFromDAO.js

```diff
@@ -116,8 +116,16 @@ foam.CLASS({
         FObject ofObject;
         try {
           ofObject = (FObject) obj;
-          delegateObject = (FObject) getTo().newInstance();
-          delegateObject = delegateObject.copyFrom(ofObject);
+
+          if ( ofObject.isPropertySet("id") ) {
+            delegateObject = (FObject) getDelegate().find(obj.getProperty("id"));
+            delegateObject = delegateObject.copyFrom(ofObject);
+          }else {
+            delegateObject = (FObject) getTo().newInstance();
+            delegateObject = delegateObject.copyFrom(ofObject);
+            //TODO the id should be undefined, that it will be set automatically
+            delegateObject.setProperty("id", new Long(0));
```

Comment:
Is this necessary? What if the id isn't a Long property?

Reply:
We need to put the id undefined that this object will be conceder as new object record to be saved automatically.
I change it to null.